compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -1166,20 +1166,16 @@ public void initPredeclaredModules(Map<Name, BPackageSymbol> predeclaredModules,
 
     @Override
     public void visit(BLangXMLNS xmlnsNode) {
-        String nsURI;
+        String nsURI = "";
         if (xmlnsNode.namespaceURI.getKind() == NodeKind.SIMPLE_VARIABLE_REF) {
             BLangSimpleVarRef varRef = (BLangSimpleVarRef) xmlnsNode.namespaceURI;
-            if (missingNodesHelper.isMissingNode(varRef.variableName.value)) {
-                nsURI = "";
-            } else {
-                // TODO: handle const-ref (#24911)
-                nsURI = "";
+            if (Symbols.isFlagOn(varRef.symbol.flags, Flags.CONSTANT)) {
+                nsURI = ((BConstantSymbol) varRef.symbol).value.toString();
+                checkInvalidNameSpaceDeclaration(xmlnsNode.pos, xmlnsNode.prefix, nsURI);
             }
         } else {
             nsURI = (String) ((BLangLiteral) xmlnsNode.namespaceURI).value;
-            if (!nullOrEmpty(xmlnsNode.prefix.value) && nsURI.isEmpty()) {
-                dlog.error(xmlnsNode.pos, DiagnosticErrorCode.INVALID_NAMESPACE_DECLARATION, xmlnsNode.prefix);
-            }
+            checkInvalidNameSpaceDeclaration(xmlnsNode.pos, xmlnsNode.prefix, nsURI);
```

Comment:
Shall we do this check after the if statement? That way, we can perform the `checkInvalidNameSpaceDeclaration` check in a single location, right after assigning a value for `nsURI`.

Reply:
This is needed in two places because we don't handle the case where the namespace uri is defined using a variable reference and which is not a constant reference. In that case also we set the namespaceURI to be empty string. We decided to handle that case in a separately (different PR). So we need to have two checks here. Otherwise an invalid error would be thrown.